roughpy_compute/common/bitmask.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef ROUGHPY_COMPUTE_COMMON_BITMASK_HPP
+#define ROUGHPY_COMPUTE_COMMON_BITMASK_HPP
+
+#include <cassert>
+#include <type_traits>
+
+namespace rpy::compute {
+
+template <typename Int>
+class BitMask
+{
+    using UInt = std::make_unsigned_t<Int>;
+    UInt base_ = 0;
+
+public:
+    constexpr BitMask() = default;
+
+    constexpr explicit BitMask(int n_bits)
+        : base_((UInt{1} << n_bits) - UInt{1})
```

Comment:
Perhaps comment here that n_bits are set to 1 by default?

Reply:
Setting a default argument would confuse the default constructor. The default sets the mask blank, and the explicit constructor sets the first n_bits bits to 1, which is the final valid mask for n_bits bits.